### ISSUE TITLE
Bug when subtracting days in CET

### ISSUE THREAD

**Issue by vetleras** [chronotope/chrono-tz#120]
The following code gives the following output.
```
let new_year = NaiveDate::from_ymd(2023, 1, 1)
  .and_hms(0, 0, 0)
  .and_local_timezone(Oslo)
  .unwrap();

println!("{}", new_year - Days::new(1));
println!("{}", new_year - Duration::days(1));
```
2022-12-30 23:00:00 CET
2022-12-31 00:00:00 CET

Is this correct behaviour? This happens for any choice of date and time.
When using Utc or London time, the outputs are equal to each other (expected behaviour in my mind).

**Comment by djc** [comment]
I think this is https://github.com/chronotope/chrono/pull/878, is that correct? If so, it should be fixed on chrono's 0.4.x branch.

**Comment by vetleras** [comment]
Great!

I am sorry that I did not see that before posting.
